source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,294 @@
+:ref:`table` type and is used on data-preparation and data-processing stages
+for:
+
+- Iterative (in general case) construction of a :ref:`table` from
+  another :ref:`tables <table>` or different in-memory :ref:`dataset`
+  representations
+
+- Construction of a :ref:`table` from different entities that hold pieces of the
+  dataset, like arrays, pointers to the memory, external entities, etc.
+
+- Changing dataset values. Since :ref:`table` is an
+  :term:`immutable <Immutability>` dataset, builder provides an ability to
+  change the values of data in the dataset under construction.
+
+- Incapsulating construction process of a :ref:`table`. This is used to hide an
+  implementation details, which (1) are unnecessary for user, (2) allow to
+  select most appropriate table implementation for every particular case.
+
+- Providing an additional information how to create a :ref:`table` inside an
+  algorithm for :ref:`results  <Result>`. This information may include metadata,
+  memory allocators need to be used, or even particular implementation of a
+  table.
+
+For details, see :ref:`table-builders` section.
+
+.. _accessor:
+
+Accessor
+--------
+
+Accessor is a concept which defines sinle way how to get the data from a
+in-memory numerical :ref:`dataset`. It allows:
+
+- To make uniform access to the data from various sets of different objects,
+  like :ref:`tables <Table>` or :ref:`table builders <table-builder>` without
+  exposing their implementation details.
+
+- To convert a variety of numeric :term:`data formats <Data format>` into a
+  smaller set (e.g. for better vectorization).
+
+- To give an ability to focus on the data acquisition in the desired :term:`data
+  format` with a wanted set of operations over the data.
+
+- To make read-only, read-write and write-only access to the data.
+
+For details, see :ref:`accessors` section.
+
+Use-case example for table, accessor and table builder
+------------------------------------------------------
+
+This section gives an basic usage scenario of :ref:`table`, :ref:`table-builder`
+and :ref:`accessor` concepts and relations between them. The following diagram
+shows objects of these concepts, which are highlighted by colors:
+
+- :ref:`table-builder` objects are blue
+
+- :ref:`table` objects are cyan
+
+- :ref:`accessors <Accessor>` are yellow
+
+- objects with gray background color are not a part of |dal_short_name|
+  specification and they are provided just for illustration purposes
+
+To perform some computations based on the dataset, one shall create a
+:ref:`table` object first. It can be done using :ref:`data-source` or
+:ref:`table-builder` objects depending on the situation. The diagram briefly
+shows a situation when :ref:`table` is created from :ref:`table-builder`
+interatively with calls of various member functions that update internal state
+of the builder. Then, the `build()` method is called which takes a snapshot of
+current builder state and creates a table from it.
```

Comment:
```suggestion
To perform computations on a dataset, one shall create a
:ref:`table` object first. It can be done using a :ref:`data-source` or
a :ref:`table-builder` object depending on the situation. The diagram briefly
shows the situation when :ref:`table` is interatively created from a :ref:`table-builder`
with calls of various member functions that update the internal state
of the builder. Then, the `build()` method is called. It takes a snapshot of the
current builder state and creates a table from it.
```